newIDE/app/src/VariablesList/ObjectInstanceVariablesDialog.js

```diff
@@ -0,0 +1,102 @@
+// @flow
+import * as React from 'react';
+import { Trans } from '@lingui/macro';
+import VariablesEditorDialog from './VariablesEditorDialog';
+import { type HotReloadPreviewButtonProps } from '../HotReload/HotReloadPreviewButton';
+import EventsRootVariablesFinder from '../Utils/EventsRootVariablesFinder';
+import { ProjectScopedContainersAccessor } from '../InstructionOrExpression/EventsScope.flow';
+import getObjectByName from '../Utils/GetObjectByName';
+
+type Props = {|
+  open: boolean,
+  project: gdProject,
+  layout: gdLayout,
+  projectScopedContainersAccessor: ProjectScopedContainersAccessor,
+  objectInstance: gdInitialInstance,
+  onApply: (selectedVariableName: string | null) => void,
+  onCancel: () => void,
+  hotReloadPreviewButtonProps?: ?HotReloadPreviewButtonProps,
+  /**
+   * If set to true, a deleted variable won't trigger a confirmation asking if the
+   * project must be refactored to delete any reference to it.
+   */
+  preventRefactoringToDeleteInstructions?: boolean,
+  initiallySelectedVariableName?: string,
+  onEditObjectVariables: () => void,
+|};
+
+const ObjectInstanceVariablesDialog = ({
+  project,
+  layout,
+  objectInstance,
+  open,
+  onCancel,
+  onApply,
+  hotReloadPreviewButtonProps,
+  preventRefactoringToDeleteInstructions,
+  initiallySelectedVariableName,
+  projectScopedContainersAccessor,
+  onEditObjectVariables,
+}: Props) => {
+  const tabs = React.useMemo(
+    () => {
+      const objectName = objectInstance.getObjectName();
+      // TODO Use projectScopedContainers to get the object
+      const variablesEditedAssociatedObject = getObjectByName(
+        project,
+        layout,
+        objectName
+      );
```

Comment:
I think it doesn't really make sense to take the point of view of a variable as they are not a code scope. For instance, here, the dialog will need to access to the object list.